node/subsystem/src/messages.rs

```diff
@@ -486,6 +499,59 @@ impl PoVDistributionMessage {
 	}
 }
 
+/// Message to the Collation Generation Subsystem.
+#[derive(Debug)]
+pub enum CollationGenerationMessage {
+	/// Initialize the collation generation subsystem
+	Initialize(CollationGenerationConfig),
+}
+
+impl CollationGenerationMessage {
+	/// If the current variant contains the relay parent hash, return it.
+	pub fn relay_parent(&self) -> Option<Hash> {
+		None
+	}
+}
+
+/// Message to the Collator Protocol subsystem
+#[derive(Debug)]
+pub enum CollatorProtocolMessage {
+	/// Signal to the collator protocol that it should connect to validators with the expectation
+	/// of collating on the given para. This is only expected to be called once, early on, if at all,
+	/// and only by the Collation Generation subsystem. As such, it will overwrite the value of
+	/// the previous signal.
+	///
+	/// This should be sent before any `DistributeCollation` message.
+	CollateOn(ParaId),
+	/// Provide a collation to distribute to validators.
+	// REVIEW: the guide is a bit inconsistent about this data type:
+	// the collation-generation page says it should have a CommittedCandidateReceipt,
+	// but the definition under Overseer Protocol types says it should just be a CandidateReceipt.
+	// I took a guess as to which of them it should be, but we should ensure I guessed right.
+	DistributeCollation(CommittedCandidateReceipt, PoV),
```

Comment:
It's not obvious to me what the difference between a `CandidateReceipt` and `CommittedCandidateReceipt` actually is; even if it's a status difference, I'd imagine one to be a newtype of the other. They're instead redundantly defined.

Reply:
The `CandidateReceipt` only stores a hash of its commitments, whereas the `CommittedCandidateReceipt` carries its commitments alongside it.

I think we want `CandidateReceipt`s here, as validators are expected to do their own validation work which will generate the commitments, that they can then compare to the hash.

We could even make a case for using `CandidateDescriptor`s instead, as I'm not 100% sure what value the commitments-hash provides here.

Reply:
In that case, should we also remove `upward_messages` and `head_data` from `struct Collation`? AFAICT, they're only useful for creating the `CandidateCommitments`.

Reply:
`struct Collation` is a v0 type, isn't it?

Reply:
No, it's a type I defined within this PR: https://github.com/paritytech/polkadot/blob/a60162e80d27c28c695df636da2a0527e63062a7/node/primitives/src/lib.rs#L289-L298

The guide suggested using a tuple, but that didn't feel great, so I made it into a struct instead.

Reply:
So, what's the verdict on this changeset:

- `DistributeCollation` takes a `CandidateDescriptor` instead of any sort of `CandidateReceipt`
- remove unneeded fields from `struct Collation`
- appropriate guide updates

Reply:
I dug into this a bit more. I'm not sure we actually want collators to distribute descriptors instead of receipts. The reason being that it complicates candidate backing. Since we issue statements based on the candidate receipt hash, it's fast to check if we've already issued a statement on a candidate receipt we are being asked by another subsystem to second. If we only got a candidate descriptor, we'd always have to do candidate validation before knowing the hash of the candidate.

But that concern isn't too strong, since there is no guarantee that the candidate receipt commitments hash we get from a collator actually matches the outputs of validation. 

 A separate refactoring could replace pretty much everything in the collation pipeline with `CandidateDescriptor`s, but note that it also requires changes in the Collation Distribution subsystem and the Candidate Validation subsystem by having `CandidateBackingMessage::Second` accept a `CandidateDescriptor`.

That all seems beyond the scope of this PR.

Reply:
Ok, opened https://github.com/paritytech/polkadot/issues/1580; we can figure this all out later.